homeassistant/config.py

```diff
@@ -1025,21 +1046,167 @@ async def merge_packages_config(
     return config
 
 
-async def async_process_component_config(  # noqa: C901
-    hass: HomeAssistant, config: ConfigType, integration: Integration
+def _get_log_message_and_stack_print_pref(
+    translation_key: str, ex: Exception, domain: str, platform_name: str
+) -> tuple[str | None, bool]:
+    """Get message to log and print stack trace preference."""
+    log_message_mapping: dict[str, tuple[str, bool]] = {
+        "component_import_err": (f"Unable to import {domain}: {ex}", False),
```

Comment:
Make constants or a strenum out of the translation keys since they are used more than once.